app/models/chef.rb

```diff
@@ -12,7 +12,22 @@ class Chef < ApplicationRecord
   validates :password, presence: true, length: { minimum: 5 }, allow_nil: true
    
   before_save { self.email = email.downcase }
-
+  before_create :set_auth_token
+  
   has_secure_password
+  has_secure_token :auth_token
+
+  private
 
+  def set_auth_token
```

Comment:
it's good that you're using `has_secure_token`. Since you're doing that, you don't need the `set_auth_token` and `generate_token`. Take them out and make sure it works without.